web/pgadmin/authenticate/oauth2.py

```diff
@@ -123,6 +123,22 @@ def login(self, form):
             [value for value in self.email_keys if value in profile.keys()]
         email = profile[email_key[0]] if (len(email_key) > 0) else None
 
+        username = email
+        username_claim = None
+        if 'OAUTH2_USERNAME_CLAIM' in self.oauth2_config:
```

Comment:
Condition should be 'OAUTH2_USERNAME_CLAIM' in self.oauth2_config[self.oauth2_current_client]. 
self.oauth2_config is value for key which maps to OAUTH2_NAME.

Reply:
Resolved this. Sorry for the late response.